gateway-spi/src/main/java/org/apache/knox/gateway/dispatch/GatewayDispatchFilter.java

```diff
@@ -145,14 +147,21 @@ private boolean isDispatchAllowed(HttpServletRequest request) {
       if (whitelist != null) {
         String requestURI = request.getRequestURI();
 
-        String decodedURL = null;
+        String decodedURL = requestURI;
         try {
           decodedURL = URLDecoder.decode(requestURI, StandardCharsets.UTF_8.name());
         } catch (UnsupportedEncodingException e) {
           //
         }
+        String baseUrl;
+        try {
+          URL url = new URL(decodedURL);
+          baseUrl = new URL(url.getProtocol(), url.getHost(), url.getPort(), "").toString();
+        } catch (MalformedURLException e) {
+          throw new RuntimeException(e);
```

Comment:
We should log the error before wrapping the exception and throwing.

Reply:
Hi @moresandeep, if we log it here then the message will appear in the log two times. I tested this with a malformed URL and the exception was still visible in the gateway.log, because at the outer level we catch and log everything. I don't see a reason to log it here too.